src/main/java/gov/nasa/pds/tools/util/LabelUtil.java

```diff
@@ -214,6 +223,84 @@ public static String getIMVersion(DOMSource source, URL context) {
      return(informationModelVersion);
   }
 
+
+  /**
+   * Get the LIDVID references in the label  (as a DOMSource)
+   * @param source The content of context as a DOMSource.
+   # @param context The location of the label being parsed from.
+   * @return lidOrLidVidReference The LID or LIDVID referenced in this label.
+   */
+  public static ArrayList<String> getLidVidReferences(DOMSource source, URL context) {
+      ArrayList<String> lidOrLidVidReferences = new ArrayList<String>(0);
+      LOG.debug("getLidVidReferences:MY_SOURCE[{}]",source);
+
+      try {
+          NodeList nodeList = (NodeList) xPathFactory.newXPath().evaluate(INTERNAL_REFERENCE_AREA,source,XPathConstants.NODESET);
+          LOG.debug("context,nodeList.getLength() {},{}",context,nodeList.getLength());
+          for (int i = 0; i < nodeList.getLength(); ++i) {
+              NodeList childList = ((Element) nodeList.item(i)).getChildNodes();
+              for (int j = 0; j < childList.getLength(); ++j) {
+                  Node node = childList.item(j);
+                  LOG.debug("node.getTextContent().trim() {}",node.getTextContent().trim());
+                  //LOG.debug("node.getTextContent().trim() {}",node.getTextContent().trim());
+                  if (node.getNodeName().equals(LIDVID_REFERENCE) || node.getNodeName().equals(LID_REFERENCE)) {
+                      lidOrLidVidReferences.add(node.getTextContent().trim());
+                  }
+              }
+         }
+     } catch (XPathExpressionException ex) {
+         LOG.error("Cannot extract field " + LIDVID_REFERENCE + " or " + LID_REFERENCE + " from context " + context.toString());
+     }
+     LOG.debug("getLidVidReferences:context,lidOrLidVidReferences {},{}",context,lidOrLidVidReferences);
+     return(lidOrLidVidReferences);
+  }
+
+  /**
+   * Get the local identifiers the label (as a DOMSource)
+   * @param source The content of context as a DOMSource.
+   # @param context The location of the label being parsed from.
+   * @return logicalIdentifiers A list of logical identifiers in this label.
+   */
+  public static ArrayList<String> getLogicalIdentifiers(DOMSource source, URL context) {
```

Comment:
@qchaupds seems like this method is almost identical to the `getLidVidReferences` method above. can we refactor this logic?

Reply:
The 2 functions have now been refactored into a 3rd common function.
Will be reflected when code is checked in.

Noted on your comment about adding new valid test when an existing one could work already.  The only disadvantage is if a particular test resource is removed in the future, check should be made to see if other tests in the feature file is not re-using it.